### PR TITLE
Homework2: Vladimir Natsevsky

### PR DESCRIPTION
**Pull request by JunglistMNSQ**

### CODE REVIEW

File: src/Homework2/codewars 1.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+There will always be at least one number in the input string.
+Output string must be two numbers separated by a single space,
+and highest number is first.l"""
+"""Test.assert_equals(high_and_low
+("4 5 29 54 4 0 -214 542 -64 1 -3 6 -6"), "542 -214");"""
+
+
+def high_and_low(numbers):
+    a = c = ''
+    i = b = t = 0
```
Comment (belovmd) [inline src/Homework2/codewars 1.py]: all these namings are completely not understandable. a c i b t

File: src/Homework2/codewars 1.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+All numbers are valid Int32, no need to validate them.
+There will always be at least one number in the input string.
+Output string must be two numbers separated by a single space,
+and highest number is first.l"""
+"""Test.assert_equals(high_and_low
+("4 5 29 54 4 0 -214 542 -64 1 -3 6 -6"), "542 -214");"""
+
+
+def high_and_low(numbers):
```
Comment (belovmd) [inline src/Homework2/codewars 1.py:19]: can it be just 
`max(string.split())`?

File: src/Homework2/codewars 2.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+15241630849, "Wrong output for 15241383936")
+
+Test.it("should return -1 for numbers which aren't perfect squares")
+Test.assert_equals(find_next_square(155), -1, "Wrong output for 155")
+Test.assert_equals(find_next_square(342786627),
+ -1, "Wrong output for 342786627")"""
+
+
+def find_next_square(sq):
+    a = sq ** 0.5
```
Comment (belovmd) [inline src/Homework2/codewars 2.py]: not clear namings

File: src/Homework2/fibonachi.py [hunk 4]
```diff
@@ -0,0 +1,11 @@
+"""- Выведите n-ое число Фибоначчи, используя только временные переменные,
+циклические операторы и условные операторы. n - вводится"""
+# первое значение последовательности 0.
+n = int(input("Введите длинну последовательности: "))
+while (n <= 0):
+    print("Число должно быть положительным")
+    n = int(input("Введите длинну последовательности: "))
+a, b, = 0, 1,
+for i in range(n - 1):
+    a, b = b, (a + b)
```
Comment (belovmd) [inline src/Homework2/fibonachi.py]: a, b = b, a+b would be enough

File: src/Homework2/longer word.py [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+Учтите что в предложении есть знаки препинания.
+Подсказки:
+my_string.split([chars]) возвращает список строк.
+len(list) - количество элементов в списке
+"""
+a = input('Введи предложение: ')
+w = 0
+W = ''
+# Знаки препинания которые ищем и исключаем, можно допольнить в любой момент.
+signs = ',.?!:;'
```
Comment (belovmd) [inline src/Homework2/longer word.py]: ][}{)( ' "  ?

File: src/Homework2/longer word.py [hunk 6]
```diff
@@ -0,0 +1,21 @@
+"""3. Найти самое длинное слово в введенном предложении.
+Учтите что в предложении есть знаки препинания.
+Подсказки:
+my_string.split([chars]) возвращает список строк.
+len(list) - количество элементов в списке
+"""
+a = input('Введи предложение: ')
+w = 0
+W = ''
```
Comment (belovmd) [inline src/Homework2/longer word.py]: incorrect naming. we use snake case

File: src/Homework2/palindrome.py [hunk 7]
```diff
@@ -0,0 +1,25 @@
+"""- Определите, является ли число палиндромом
+(читается слева направо и справа налево одинаково).
+Число положительное целое, произвольной длины.
+Задача требует работать только с числами (без конвертации числа в строку)"""
+a = int(input("Введите целое положительное число: "))
+while (int(a) <= 0):
+    a = int(input("Попробуйте еще раз: "))
+b = n = m = 0
```
Comment (belovmd) [inline src/Homework2/palindrome.py]: bad namings

File: src/Homework2/palindrome.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+c = a
+# Определяем количество разрядов
+while (a // (10 ** n)) != 0:
+    n += 1
+n -= 1
+# Немного магии
+while n >= 0:
```
Comment (belovmd) [inline src/Homework2/palindrome.py]: one while for this task would be enough

File: src/Homework2/price.py [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    else:
+        i += 1
+# Найденные числа делаем численными =)
+N = int(N)
+M = int(M)
+L = int(L)
+# Копейки переводим в сотые и складываем с рублями,
+# умножаем на кол-во
+S = (N + (M * 0.01)) * L
+S = str(S)
```
Comment (belovmd) [inline src/Homework2/price.py]: why do we need string here?

**Review by belovmd — COMMENTED** [review]
Please address these issues

File: src/Homework2/codewars 1.py (hunk 1 above)
Comment (JunglistMNSQ) [reply src/Homework2/codewars 1.py]: complete

File: src/Homework2/codewars 1.py (hunk 2 above)
Comment (JunglistMNSQ) [reply src/Homework2/codewars 1.py:19]: complete

File: src/Homework2/codewars 2.py (hunk 3 above)
Comment (JunglistMNSQ) [reply src/Homework2/codewars 2.py]: complete

File: src/Homework2/fibonachi.py (hunk 4 above)
Comment (JunglistMNSQ) [reply src/Homework2/fibonachi.py]: complete

File: src/Homework2/longer word.py (hunk 5 above)
Comment (JunglistMNSQ) [reply src/Homework2/longer word.py]: complete

File: src/Homework2/longer word.py (hunk 6 above)
Comment (JunglistMNSQ) [reply src/Homework2/longer word.py]: complete

File: src/Homework2/palindrome.py (hunk 7 above)
Comment (JunglistMNSQ) [reply src/Homework2/palindrome.py]: complete

File: src/Homework2/palindrome.py (hunk 8 above)
Comment (JunglistMNSQ) [reply src/Homework2/palindrome.py]: complete

File: src/Homework2/price.py (hunk 9 above)
Comment (JunglistMNSQ) [reply src/Homework2/price.py]: complete

**Review by belovmd — COMMENTED** [review]
9

**Pull request closed without merging**